src/test/groovy/com/pszymczyk/consul/infrastructure/ConsulWaiterTest.groovy

```diff
@@ -14,4 +18,80 @@ class ConsulWaiterTest extends Specification {
         def ex = thrown EmbeddedConsulException
         ex.message =~ "Could not start Consul*"
     }
+
+
+    def "should use token to get list of registered nodes"() {
+        when:
+        def uuid = UUID.randomUUID().toString()
+        def conf = """{
+          "datacenter": "us",
+          "acl_datacenter": "us",
+          "acl_default_policy": "deny",
+          "acl_agent_master_token": ${uuid},
+          "acl_master_token": ${uuid},
+          "acl_agent_token": ${uuid},
+          "acl_token": ${uuid},
+          "acl_enable_key_list_policy": true,
+          "acl_down_policy": "deny"
+        }"""
+
+        ConsulProcess consul = ConsulStarterBuilder.consulStarter()
+                .withToken(uuid)
+                .withCustomConfig(conf)
+                .withConsulVersion("1.0.0")
+                .build().start()
+
+        then:
+        noExceptionThrown()
+
+        cleanup:
+        consul.close()
+    }
+
+    def "unable to get registered nodes with wrong token"() {
+        when:
+        def token = UUID.randomUUID().toString()
+        def wrongToken = UUID.randomUUID().toString()
+        def conf = """{
+          "datacenter": "us",
+          "acl_datacenter": "us",
+          "acl_default_policy": "deny",
+          "acl_master_token": ${token},
+          "acl_enable_key_list_policy": true,
+          "acl_down_policy": "deny"
+        }"""
+
+        ConsulStarterBuilder.consulStarter()
+                .withToken(wrongToken)
+                .withCustomConfig(conf)
+                .withConsulVersion("1.0.0")
+                .withWaitTimeout(3)
+                .build().start()
+
+        then:
+        def ex = thrown EmbeddedConsulException
+        ex.message =~ "Could not start Consul*"
+
+    }
+
+    def "should get registered nodes with token for consul without token"() {
+        when:
+        def token = UUID.randomUUID().toString()
+        def conf = """{
+          "datacenter": "us",
+        }"""
+
+        ConsulProcess consul = ConsulStarterBuilder.consulStarter()
+                .withToken(token)
+                .withCustomConfig(conf)
+                .withConsulVersion("1.0.0")
+                .build().start()
+
+        then:
+        noExceptionThrown()
```

Comment:
again the test name do not correspond with then section - you don't check is any node registered here